### PR TITLE
Update condition LastUpdateTime when  kubectl certificate approve/deny --force operation 

### LINKED ISSUE #91868: kubectl certificate approve --force has no effect
**What happened**:

```
return o.modifyCertificateCondition(o.builder, o.clientSet, force, func(csr *certificatesv1beta1.CertificateSigningRequest) (*certificatesv1beta1.CertificateSigningRequest, bool) {
		var alreadyApproved bool
		for _, c := range csr.Status.Conditions {
			if c.Type == certificatesv1beta1.CertificateApproved {
				alreadyApproved = true
			}
		}
		if alreadyApproved {
			return csr, true
		}
		csr.Status.Conditions = append(csr.Status.Conditions, certificatesv1beta1.CertificateSigningRequestCondition{
			Type:           certificatesv1beta1.CertificateApproved,
			Reason:         "KubectlApprove",
			Message:        "This CSR was approved by kubectl certificate approve.",
			LastUpdateTime: metav1.Now(),
		})
		return csr, false
	})
```

if approved is true.the csr status conditions will not change。
```
if !hasCondition || force {
				csr, err = clientSet.CertificateSigningRequests().UpdateApproval(csr)
				if errors.IsConflict(err) && i < 10 {
					if err := info.Get(); err != nil {
						return err
					}
					continue
				}
				if err != nil {
					return err
				}
			}
```
even if run UpdateApproval,but the csr had not changeed so，no update will be applied

kubectl certificate deny has the same problem。

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
[root@master01 ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2020-05-12T09:28:18Z", GoVersion:"go1.12.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by wawa0210**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #91868

**Special notes for your reviewer**:

At present, if the certificate has been approved or deny, and then execute `kubectl certificate approve |deny --force` again, the condition recorded will not be ingored . But after `--force`, it should try to update and the condition should be recorded

**Does this PR introduce a user-facing change?**:

```release-note
kubectl certificate approve --force operation will update the condition 
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
kubectl certificate approve --force operation will update the condition
```

/assign @liggitt

### CODE REVIEW

**Comment by liggitt** [comment]
It's not clear what this is fixing… checking an attribute on the ranged variable is equivalent. There's also no test demonstrating the issue and fix. 

https://github.com/kubernetes/kubernetes/pull/91778 is also in progress to change these commands… let's get that in first, then consider whether this needed

File: staging/src/k8s.io/kubectl/pkg/cmd/certificates/certificates.go [hunk 1]
```diff
@@ -146,11 +146,14 @@ func NewCmdCertificateApprove(f cmdutil.Factory, ioStreams genericclioptions.IOS
...
-	return o.modifyCertificateCondition(o.builder, o.clientSet, force, func(csr *certificatesv1beta1.CertificateSigningRequest) (*certificatesv1beta1.CertificateSigningRequest, bool) {
+	return o.modifyCertificateCondition(o.builder, o.clientSet, force, func(csr *certificatesv1beta1.CertificateSigningRequest, force bool) (*certificatesv1beta1.CertificateSigningRequest, bool) {
 		var alreadyApproved bool
-		for _, c := range csr.Status.Conditions {
-			if c.Type == certificatesv1beta1.CertificateApproved {
+		for i := range csr.Status.Conditions {
+			if csr.Status.Conditions[i].Type == certificatesv1beta1.CertificateApproved {
 				alreadyApproved = true
+				if force {
+					csr.Status.Conditions[i].LastUpdateTime = metav1.Now()
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/certificates/certificates.go:155]: I wouldn't expect this… we're not changing the condition so we shouldn't change the timestamp

Comment (wawa0210) [reply staging/src/k8s.io/kubectl/pkg/cmd/certificates/certificates.go:155]: When approve|deny --force, kubectl will request the apiserver to make changes

image

So, I think it makes sense to update the LastUpdateTime of the condition at this time

**Comment by wawa0210** [comment]
> It's not clear this change is correct. There's also no test demonstrating the issue and fix.

Sorry, I didn't notice that you are refactoring the `csr` code before. I will first wait for your pr to be merged, and then add test code to this logic, if this pr is needed

**Comment by liggitt** [comment]
`--force` sends a PUT to the server correctly. Since the updated content is identical to what is already persisted, there is no change, but this is working as intended.

/close

**Comment by wawa0210** [comment]
> `--force` sends a PUT to the server correctly. Since the updated content is identical to what is already persisted, there is no change, but this is working as intended.
> 

Got it, 
I have a small question, if we use `--force`, although a put request is sent to apiserver, there will be no changes. What is the usage scenario of `--force` here? Is it necessary to exist

**Comment by liggitt** [comment]
> I have a small question, if we use `--force`, although a put request is sent to apiserver, there will be no changes. What is the usage scenario of `--force` here?

That's a good question. I don't really know.

> Is it necessary to exist

Probably not, but dropping the flag is more trouble than it's worth at this point. It doesn't harm anything.

**Comment by wawa0210** [comment]
> Probably not, but dropping the flag is more trouble than it's worth at this point. It doesn't harm anything.

I used to deviate from the block understanding. I thought that the status of csr can be changed bidirectionally from approved <-->deny, so in this case we might need `--force`, and now it seems that csr can only be approved Or deny, once the status is determined, continue to change is not supported

**Pull request closed without merging**